app/questionnaire/questionnaire_store_updater.py

```diff
@@ -321,6 +321,46 @@ def _capture_section_dependencies_for_answer(self, answer_id: str) -> None:
             else:
                 self.dependent_sections.add(DependentSection(section_id, None, None))
 
+    def _capture_section_dependencies_progress_value_source_for_current_section(
+        self,
+    ) -> None:
+        """
+        Captures a unique list of section ids that are dependents of the current section, for progress value sources.
+        """
+        current_section_id = self._current_location.section_id
+        progress_deps_mapping = (
+            self._schema.when_rules_section_dependencies_by_section_for_progress_value_source
+        )
+        for section_dependency_id in progress_deps_mapping:
+            if current_section_id in progress_deps_mapping[section_dependency_id]:
```

Comment:
We should flip the dependency capture, at the moment, we are going through every section until the section is found or we run out of sections. The lookup should be by current section/block, so you don't have to iterate through all sections that have a dependency. All other dependency work is lookup by current section/block/answer for performance reasons.

Reply:
Updated this 👍